app/src/main/java/com/example/reduxtestapp/redux/middleware/RootMiddleware.kt

```diff
@@ -0,0 +1,84 @@
+package com.example.reduxtestapp.redux.middleware
+
+import com.example.reduxtestapp.redux.Action
+import com.example.reduxtestapp.redux.AppState
+import com.example.reduxtestapp.redux.MiddlewareResult
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import org.reduxkotlin.Dispatcher
+import org.reduxkotlin.Store
+
+class RootMiddleware(
+    private val dispatcher: CoroutineDispatcher,
+    countryMiddleware: CountryMiddleware,
+    priceMiddleware: PriceMiddleware,
+    todoMiddleware: TodoMiddleware
+) {
+
+    private val middlewareList = listOf(
+        countryMiddleware,
+        priceMiddleware,
+        todoMiddleware
+    )
```

Comment:
A bit of miscommunication here maybe, RootMiddleware was just supposed to be a MiddlewareResult handler and not an aggregator of every middleware. Let's discuss again